packages/realm/bindgen/js_spec.yml

```diff
@@ -17,3 +17,12 @@ classes:
       get_cpu_arch: () -> std::string
       # print: (const char* fmt, ...) # can't expose varargs directly. Could expose a fixed overload.
 
+  WeakSyncSession:
+    cppName: std::weak_ptr<SyncSession>
+    constructors:
+      weakCopyOf: '(shared: SharedSyncSession)'
+    methods:
+      rawDereference:
+        sig: '() const -> Nullable<SharedSyncSession>'
+        cppName: lock
```

Comment:
None of these are directly used by code. Both are only used by the friendlier helpers in `binding.ts` at the very bottom of this PR.

Reply:
If they're not intended for consumption by the SDK, could we find a way to declare that? Perhaps in the `binding.ts` by marking them as deprecated?

Reply:
> like the pattern of using `binding.ts` to augment the raw C++ APIs from TS.

I agree, that seems best.

Reply:
I didn't see a good technique that would let me add annotations or comments to both the static and instance methods. If you care, and are able to find such a technique, we can always add it later.